src/Datadog.Trace.ServiceFabric/Remoting.cs

```diff
@@ -0,0 +1,373 @@
+
+            try
+            {
+                scope = Tracer.Instance.ActiveScope;
+
+                if (scope != null &&
+                    e is ServiceRemotingFailedResponseEventArgs failedResponseArg &&
+                    failedResponseArg.Error != null)
+                {
+                    scope.Span?.SetException(failedResponseArg.Error);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error accessing active scope or setting error tags.");
+            }
+            finally
+            {
+                scope?.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Event handler called when the Service Remoting server receives an incoming request.
+        /// </summary>
+        /// <param name="sender">The object that raised the event.</param>
+        /// <param name="e">The event arguments.</param>
+        private static void ServiceRemotingServiceEvents_ReceiveRequest(object? sender, EventArgs e)
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            GetMessageHeaders(e, out var eventArgs, out var messageHeaders);
+            PropagationContext propagationContext = default;
+            SpanContext? spanContext = null;
+
+            // extract propagation context from message headers for distributed tracing
+            if (messageHeaders != null)
+            {
+                propagationContext = ExtractContext(messageHeaders);
+
+                if (propagationContext.TraceId > 0 && propagationContext.ParentSpanId > 0)
+                {
+                    spanContext = new SpanContext(propagationContext.TraceId, propagationContext.ParentSpanId, (SamplingPriority?)propagationContext.SamplingPriority);
+                }
+            }
+
+            var tracer = Tracer.Instance;
+            var span = CreateSpan(tracer, spanContext, SpanKinds.Server, eventArgs, messageHeaders);
+
+            if (!string.IsNullOrEmpty(propagationContext.Origin))
+            {
+                span.SetTag(Tags.Origin, propagationContext.Origin);
+            }
+
+            tracer.ActivateSpan(span);
+        }
+
+        /// <summary>
+        /// Event handler called when the Service Remoting server sends a response
+        /// after processing an incoming request.
+        /// </summary>
+        /// <param name="sender">The object that raised the event.</param>
+        /// <param name="e">The event arguments.</param>
+        private static void ServiceRemotingServiceEvents_SendResponse(object? sender, EventArgs e)
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            // var successfulResponseArg = e as ServiceRemotingResponseEventArgs;
+            // var failedResponseArg = e as ServiceRemotingFailedResponseEventArgs;
+
+            var scope = Tracer.Instance.ActiveScope;
+
+            if (scope != null)
```

Comment:
Is there any reason there's less error handling here than in the client side when it attempts to close the active span?

Reply:
Oops, oversight. I added more error handling f3e0bc7f.